src/coreclr/vm/exceptionhandling.cpp

```diff
@@ -7159,5 +7431,842 @@ void ExceptionTracker::ResetThreadAbortStatus(PTR_Thread pThread, CrawlFrame *pC
+            {
+                ETW::ExceptionLog::ExceptionFinallyEnd();
+                EEToProfilerExceptionInterfaceWrapper::ExceptionUnwindFinallyLeave();
+            }
+            else
+            {
+                ETW::ExceptionLog::ExceptionCatchEnd();
+                ETW::ExceptionLog::ExceptionThrownEnd();
+                EEToProfilerExceptionInterfaceWrapper::ExceptionCatcherLeave();
+            }
+        }
+        // TODO: add the m_EHClauseInfo for profiler 
+        //m_EHClauseInfo.ResetInfo();
+    }
+}
+
+extern "C" void * QCALLTYPE RhpCallCatchFunclet(QCall::ObjectHandleOnStack exceptionObj, BYTE* pHandlerIP, REGDISPLAY* pvRegDisplay, ExInfo* exInfo)
+{
+    QCALL_CONTRACT;
+
+    BEGIN_QCALL;
+    GCX_COOP_NO_DTOR();
+
+    Thread* pThread = GET_THREAD();
+    Frame* pFrame = pThread->GetFrame();
+    MarkInlinedCallFrameAsFuncletCall(pFrame);
+    HandlerFn* pfnHandler = (HandlerFn*)pHandlerIP;
+    exInfo->m_sfHighBound = exInfo->m_frameIter.m_crawl.GetRegisterSet()->SP;
+    DWORD_PTR dwResumePC;
+
+    if (pHandlerIP != NULL)
+    {
+        _ASSERTE(exInfo->m_sfCallerOfActualHandlerFrame == EECodeManager::GetCallerSp(pvRegDisplay));
+        OBJECTREF throwable = exceptionObj.Get();
+        throwable = PossiblyUnwrapThrowable(throwable, exInfo->m_frameIter.m_crawl.GetAssembly());
+
+        UINT_PTR establisherFrame = GetEstablisherFrame(pvRegDisplay, exInfo);
+        
+        exInfo->m_csfEHClause = CallerStackFrame((UINT_PTR)GetCurrentSP());
+        exInfo->m_csfEnclosingClause = CallerStackFrame::FromRegDisplay(exInfo->m_frameIter.m_crawl.GetRegisterSet());
+
+        MethodDesc *pMD = exInfo->m_frameIter.m_crawl.GetFunction();
+        // Profiler, debugger and ETW events
+        MakeCallbacksRelatedToHandler(true, pThread, pMD, &exInfo->m_ClauseForCatch, (DWORD_PTR)pHandlerIP, GetSP(pvRegDisplay->pCurrentContext));
+
+#ifdef USE_FUNCLET_CALL_HELPER
+        // Invoke the catch funclet. 
+        // Since the actual caller of the funclet is the assembly helper, pass the reference
+        // to the CallerStackFrame instance so that it can be updated.
+        CallerStackFrame* pCallerStackFrame = &exInfo->m_csfEHClause;
+        UINT_PTR *pFuncletCallerSP = &(pCallerStackFrame->SP);
+        dwResumePC = CallEHFunclet(OBJECTREFToObject(throwable),
+                                   CastHandlerFn(pfnHandler),
+                                   GetFirstNonVolatileRegisterAddress(pvRegDisplay->pCurrentContext),
+                                   pFuncletCallerSP);
+#else
+        dwResumePC = pfnHandler(establisherFrame, OBJECTREFToObject(throwable));
+#endif            
+        // Profiler, debugger and ETW events
+        MakeCallbacksRelatedToHandler(false, pThread, pMD, &exInfo->m_ClauseForCatch, (DWORD_PTR)pHandlerIP, GetSP(pvRegDisplay->pCurrentContext));
+        SetIP(pvRegDisplay->pCurrentContext, dwResumePC);
+    }
+
+    ULONG64 targetSp = GetSP(pvRegDisplay->pCurrentContext);
```

Comment:
```suggestion
    UINT_PTR targetSp = GetSP(pvRegDisplay->pCurrentContext);
```
ULONG64 does not make sense on 64-bit here